codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -211,16 +210,16 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
             final Http2Headers headers, final int streamDependency, final short weight,
             final boolean exclusive, final int padding, final boolean endOfStream,
             final ChannelPromise promise) {
-        Http2Stream stream = connection.stream(streamId);
-        ChannelFuture lastDataWrite = stream != null ? flowController().lastFlowControlledFrameSent(stream) : null;
+        final Http2Stream stream;
         try {
             if (connection.isGoAway()) {
                 throw connectionError(PROTOCOL_ERROR, "Sending headers after connection going away.");
             }
-
-            if (stream == null) {
+            Http2Stream tmpStream = connection.stream(streamId);
```

Comment:
Allow use of final for stream above which avoids the later lookup by stream id in the listener.